src/Form/Question.php

```diff
@@ -97,6 +97,16 @@ public function getQuestionType(): ?QuestionTypeInterface
         return new $type();
     }
 
+    /**
+     * Get the extra datas for the question.
+     *
+     * @return ?array
+     */
+    public function getExtraDatas(): ?array
+    {
+        return json_decode($this->fields['extra_data'] ?? "[]", true);
```

Comment:
Same here, maybe we need a `validateExtraDataValues` method in the `QuestionTypeInterface` that make sure we don't have any invalid values here and that any mandatory extra param is set.

If the method fails, the question should not be displayed in the form displayed to the end user and an error should be logged.